src/main/java/com/adtsw/jcommons/jmx/JmxMetricProviderService.java

```diff
@@ -0,0 +1,270 @@
+package com.adtsw.jcommons.jmx;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.JMException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeDataSupport;
+import java.io.Closeable;
+import java.lang.management.ManagementFactory;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A Class to get all the JMX Metrics from JMX MBeanServer. Also this class continuously monitor the MBeanServer
+ * with the given period and watch for any changes in metrics. It notify metrics add or delete events to
+ * {@link JmxMetricsListener JmxMetricsListener}.
+ */
+public class JmxMetricProviderService implements Closeable {
```

Comment:
Does it make to extract a blue print (an interface) out of this? I think it might help in understanding the expectations of this class